src/libcore/option.rs

```diff
@@ -858,6 +860,8 @@ impl<T> Option<T> {
 
         match *self {
             Some(ref mut v) => v,
+            // SAFETY: a `None` variant for `self` would have been replaced by a `Some`
+            // variant in the code above.
             None => unsafe { hint::unreachable_unchecked() },
         }
```

Comment:
Huh. Nothing to change here, but I'm surprised we can't write this with a binding on the None like `ref mut this @ None => *this = Some(f());` or so here. Probably a borrow check limitation?

Reply:
How would that help? Your `this` is just a reborrow of `self`, but we need to reference the value inside `Some`.

Reply:
Ah, I see. Yeah, makes sense.